### PR TITLE
fix_: ignore log partial API request

### PR DESCRIPTION
**Pull request by qfrank**
Major changes:
- Added extra sensitive fields in regex.

- Added some new V2 APIs and removed logging for the old ones:
```
VerifyDatabasePassword -> VerifyDatabasePasswordV2
MigrateKeyStoreDir -> MigrateKeyStoreDirV2
DeleteImportedKey -> DeleteImportedKeyV2
SignTypedData -> SignTypedDataV2
```

- Removed the following sensitive APIs for logging as a quick fix, we can create a separate pull request to create V2 APIs for them later if needed:
```
SignTypedDataV4 
SendTransactionWithChainID 
SendTransaction 
ValidateMnemonic
ExportUnencryptedDatabase
ImportUnencryptedDatabase
ChangeDatabasePassword
ConvertToKeycardAccount
ConvertToRegularAccount
```

Relate [comment](https://github.com/status-im/status-go/pull/5812#issuecomment-2364600859)

Please review the code by commit , tx

### CODE REVIEW

**Comment by qfrank** [comment]
cc @churik

File: mobile/status.go [hunk 1]
```diff
@@ -281,22 +281,55 @@ func verifyAccountPassword(keyStoreDir, address, password string) string {
...
+	if err != nil {
+		return makeJSONResponse(err)
+	}
+	err = statusBackend.VerifyDatabasePassword(request.KeyUID, request.Password)
+	return makeJSONResponse(err)
 }
 
 // verifyDatabasePassword verifies database password.
-func verifyDatabasePassword(keyUID, password string) string {
+// Deprecated use VerifyDatabasePasswordV2 instead
```
Comment (igor-sirotin) [inline mobile/status.go]:
```suggestion
// Deprecated: use VerifyDatabasePasswordV2 instead
```

File: mobile/status.go [hunk 2]
```diff
@@ -737,12 +795,8 @@ func sendTransactionWithChainID(chainID int, txArgsJSON, password string) string
 	return prepareJSONResponseWithCode(hash.String(), err, code)
 }
 
+// SendTransaction converts RPC args and calls backend.SendTransaction.
 func SendTransaction(txArgsJSON, password string) string {
-	return logAndCallString(sendTransaction, txArgsJSON, password)
```
Comment (igor-sirotin) [inline mobile/status.go]: Let's not nightmare the git history there and back 😄 
Keep the private func and the public wrapper for overall consistency in the file.
```suggestion
    return sendTransaction(txArgsJSON, password)
```

Same for other cases in this file.

File: mobile/status_request_log.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/status-im/status-go/logutils/requestlog"
 )
 
-var sensitiveRegex = regexp.MustCompile(`(?i)(".*?(password|mnemonic|openseaAPIKey|poktToken|alchemyArbitrumMainnetToken|raribleTestnetAPIKey|alchemyOptimismMainnetToken|statusProxyBlockchainUser|alchemyEthereumSepoliaToken|alchemyArbitrumSepoliaToken|infuraToken|raribleMainnetAPIKey|alchemyEthereumMainnetToken).*?")\s*:\s*("[^"]*")`)
+var sensitiveRegex = regexp.MustCompile(`(?i)(".*?(password|mnemonic|openseaAPIKey|poktToken|alchemyArbitrumMainnetToken|raribleTestnetAPIKey|alchemyOptimismMainnetToken|statusProxyBlockchainUser|alchemyEthereumSepoliaToken|alchemyArbitrumSepoliaToken|infuraToken|raribleMainnetAPIKey|alchemyEthereumMainnetToken|alchemyOptimismSepoliaToken|verifyENSURL|verifyTransactionURL).*?")\s*:\s*("[^"]*")`)
```
Comment (igor-sirotin) [inline mobile/status_request_log.go]: This is becoming very unreadable now.
Do you think we could re-write this with something like this?

```suggestion
var sensitiveKeys = []string{
	"password",
	"mnemonic",
	"openseaAPIKey",
	"poktToken",
	"alchemyArbitrumMainnetToken",
	"raribleTestnetAPIKey",
	"alchemyOptimismMainnetToken",
	"statusProxyBlockchainUser",
	"alchemyEthereumSepoliaToken",
	"alchemyArbitrumSepoliaToken",
	"infuraToken",
	"raribleMainnetAPIKey",
	"alchemyEthereumMainnetToken",
	"alchemyOptimismSepoliaToken",
	"verifyENSURL",
	"verifyTransactionURL",
}

var sensitiveRegexString = fmt.Sprintf(`(?i)(".*?(%s).*?")\s*:\s*("[^"]*")`, strings.Join(sensitiveKeys, "|"))
var sensitiveRegex = regexp.MustCompile(sensitiveRegexString)
```
```

File: protocol/requests/delete_imported_key.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		return ErrDeleteImportedKeyEmptyAddress
+	}
+	if r.Password == "" {
+		return ErrDeleteImportedKeyEmptyPassword
+	}
+	if r.KeyStoreDir == "" {
+		return ErrDeleteImportedKeyEmptyKeyStoreDir
+	}
+	return nil
+}
```
Comment (igor-sirotin) [inline protocol/requests/delete_imported_key.go]: Would be nice of us to start using [go-validator](https://github.com/go-validator/validator).
We use it for some structs, but usually we write such funcs 🙁

Comment (qfrank) [reply protocol/requests/delete_imported_key.go]: the whole file was generated by cursor, I didn't write one... that's what AI learnt from our codebase(kidding), actually it suggested using go-validator first, but I just want to keep the same code style as before, maybe we can create a specific refactor on this topic?

Comment (igor-sirotin) [reply protocol/requests/delete_imported_key.go]: > but I just want to keep the same code style as before, maybe we can create a specific refactor on this topic?

We can't afford to refactor the whole status-go in one step. 
The only way I see it working with introducing such things to code is:
1. Agree on the decision
2. Start writing new code with the manner
3. When touching the old code, change it

Comment (igor-sirotin) [reply protocol/requests/delete_imported_key.go]: Also, we use validator in some structs, so it's not the questions about "keeping same code style as before". It's already a mess 🙂 

https://github.com/status-im/status-go/blob/8cf4feec1e18870aa0007fa8123375bbdcf2712e/params/config.go#L332 https://github.com/status-im/status-go/blob/f660be0daa7cc742a07f6a139ea5ac9966f3ebe0/server/pairing/config.go#L15

File: mobile/status.go (hunk 2 above)
Comment (qfrank) [reply mobile/status.go]: fixed😅

File: mobile/status_request_log.go (hunk 3 above)
Comment (qfrank) [reply mobile/status_request_log.go]: fixed

File: mobile/status.go (hunk 1 above)
Comment (qfrank) [reply mobile/status.go]: fixed

File: protocol/requests/delete_imported_key.go (hunk 4 above)
Comment (qfrank) [reply protocol/requests/delete_imported_key.go]: fixed🙏

**Review by igor-sirotin — APPROVED** [review]
Thank you for the changes and the tests 👍

**Review by osmaczko — APPROVED** [review]

**Pull request merged**